tests/components/gdacs/test_config_flow.py

```diff
@@ -0,0 +1,130 @@
+    FEED,
+    async_setup_entry,
+    async_unload_entry,
+    config_flow,
+)
+from homeassistant.const import (
+    CONF_LATITUDE,
+    CONF_LONGITUDE,
+    CONF_RADIUS,
+    CONF_SCAN_INTERVAL,
+    CONF_UNIT_SYSTEM,
+)
+
+from tests.common import MockConfigEntry
+
+
+@pytest.fixture
+def config_entry():
+    """Create a mock GDACS config entry."""
+    return MockConfigEntry(
+        domain=DOMAIN,
+        data={
+            CONF_LATITUDE: -41.2,
+            CONF_LONGITUDE: 174.7,
+            CONF_RADIUS: 25,
+            CONF_UNIT_SYSTEM: "metric",
+            CONF_SCAN_INTERVAL: 300.0,
+            CONF_CATEGORIES: [],
+        },
+        title="-41.2, 174.7",
+    )
+
+
+async def test_duplicate_error(hass, config_entry):
+    """Test that errors are shown when duplicates are added."""
+    conf = {CONF_LATITUDE: -41.2, CONF_LONGITUDE: 174.7, CONF_RADIUS: 25}
+
+    config_entry.add_to_hass(hass)
+    flow = config_flow.GdacsFlowHandler()
+    flow.hass = hass
+
+    result = await flow.async_step_user(user_input=conf)
+    assert result["errors"] == {"base": "identifier_exists"}
+
+
+async def test_show_form(hass):
+    """Test that the form is served with no input."""
+    flow = config_flow.GdacsFlowHandler()
+    flow.hass = hass
+
+    result = await flow.async_step_user(user_input=None)
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["step_id"] == "user"
+
+
+async def test_step_import(hass):
+    """Test that the import step works."""
+    conf = {
+        CONF_LATITUDE: -41.2,
+        CONF_LONGITUDE: 174.7,
+        CONF_RADIUS: 25,
+        CONF_UNIT_SYSTEM: "metric",
+        CONF_SCAN_INTERVAL: timedelta(minutes=4),
+        CONF_CATEGORIES: ["Drought", "Earthquake"],
+    }
+
+    flow = config_flow.GdacsFlowHandler()
+    flow.hass = hass
+
+    result = await flow.async_step_import(import_config=conf)
+    assert result["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
+    assert result["title"] == "-41.2, 174.7"
+    assert result["data"] == {
+        CONF_LATITUDE: -41.2,
+        CONF_LONGITUDE: 174.7,
+        CONF_RADIUS: 25,
+        CONF_UNIT_SYSTEM: "metric",
+        CONF_SCAN_INTERVAL: 240.0,
+        CONF_CATEGORIES: ["Drought", "Earthquake"],
+    }
+
+
+async def test_step_user(hass):
+    """Test that the user step works."""
+    hass.config.latitude = -41.2
+    hass.config.longitude = 174.7
+    conf = {CONF_RADIUS: 25}
+
+    flow = config_flow.GdacsFlowHandler()
+    flow.hass = hass
+
+    result = await flow.async_step_user(user_input=conf)
+    assert result["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
+    assert result["title"] == "-41.2, 174.7"
+    assert result["data"] == {
+        CONF_LATITUDE: -41.2,
+        CONF_LONGITUDE: 174.7,
+        CONF_RADIUS: 25,
+        CONF_UNIT_SYSTEM: "metric",
+        CONF_SCAN_INTERVAL: 300.0,
+        CONF_CATEGORIES: [],
+    }
+
+
+async def test_component_unload_config_entry(hass, config_entry):
```

Comment:
Please move this test to a module `test_init.py`.

Reply:
OK, had to introduce `conftest.py` as well for a shared fixture.